Description: Work around python bug #1652
 This avoids GAP printing extraneous "gzip: stdout: Broken pipe" errors
 However it is NOT related to Debian bug #844789
Author: Ximin Luo <infinity0@debian.org>
Bug: https://bugs.python.org/issue1652
Forwarded: https://trac.sagemath.org/ticket/22440
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/src/sage/interfaces/expect.py
+++ b/sage/src/sage/interfaces/expect.py
@@ -42,6 +42,7 @@
 from __future__ import absolute_import
 
 import os
+import signal
 import sys
 import weakref
 import time
@@ -479,6 +480,7 @@
                         timeout=None,  # no timeout
                         env=pexpect_env,
                         name=self._repr_(),
+                        preexec_fn=lambda: signal.signal(signal.SIGPIPE, signal.SIG_DFL),
                         quit_string=self._quit_string())
             except (ExceptionPexpect, pexpect.EOF) as e:
                 # Change pexpect errors to RuntimeError
